### PR TITLE
Check binary compatibility before releasing 0.10.5

### PR DESCRIPTION
**Pull request by danimontoya**
Check binary compatibility

### CODE REVIEW

File: arrow-fx/build.gradle [hunk 1]
```diff
@@ -9,6 +9,10 @@ plugins {
     id "ru.vyarus.animalsniffer"
 }
 
+apiValidation {
```
Comment (rachelcarmena) [inline arrow-fx/build.gradle]: Good job @danimontoya :clap: Just a comment. Could `apiValidation` section be located in `build.gradle`?

Comment (danimontoya) [reply arrow-fx/build.gradle]: I guess you mean in the root `build.gradle`? I've been playing with it and no, I couldn't =\
so I had to add it to arrow-fx and worked as I was expecting if I would have added it to the root build.gradle

Comment (danimontoya) [reply arrow-fx/build.gradle]: you right @rachelcarmena ! just moved it to the root one! thanks!

**Comment by rachelcarmena** [comment]
> LGTM although we should create a ticket to add the api check change to the CI

Or maybe it can be included in this PR. Just add a new step in `.github/workflows/build_arrow-fx.yml` at the end:

```
    - name: Check the API
      run: ./gradlew apiCheck
```

**Comment by danimontoya** [comment]
Oh thats awesome! But adding it now can create some issues.. its better to create a separate issue and tackle it together with adding the possible ignore projects and ignore packages!!

**Comment by rachelcarmena** [comment]
> Oh thats awesome! But adding it now can create some issues.. its better to create a separate issue and tackle it together with adding the possible ignore projects and ignore packages!!

Good idea!!! :raised_hands:

**Comment by danimontoya** [comment]
Task will be continued here: https://github.com/arrow-kt/arrow-fx/issues/106

**Pull request merged**